openpype/hosts/nuke/api/lib.py

```diff
@@ -2114,12 +2141,87 @@ def set_root_colorspace(self, imageio_host):
 
 Reopening Nuke should synchronize these paths and resolve any discrepancies.
 """
-                nuke.message(
-                    message.format(
-                        env_path=current_ocio_path,
-                        settings_path=config_data["path"]
-                    )
+            nuke.message(
+                message.format(
+                    env_path=current_ocio_path,
+                    settings_path=config_data["path"]
                 )
+            )
+            return True
+
+    def _set_ocio_config_path_to_workfile(self, config_data):
+        """ Set OCIO config path to workfile
+
+        Path set into nuke workfile. It is trying to replace path with
+        environment variable if possible. If not, it will set it as it is.
+        It also saves the script to apply the change, but only if it's not
+        empty Untitled script.
+
+        Args:
+            config_data (dict): OCIO config data from settings
+
+        """
+        # replace path with env var if possible
+        ocio_path = self._replace_ocio_path_with_env_var(
+            config_data["path"]
+        )
+        log.info("Setting OCIO config path to: `{}`".format(
+            ocio_path))
+
+        self._root_node["customOCIOConfigPath"].setValue(
+            ocio_path
+        )
+        self._root_node["OCIO_config"].setValue("custom")
+
+        # only save script if it's not empty
+        if self._root_node["name"].value() != "":
+            log.info("Saving script to apply OCIO config path change.")
+            nuke.scriptSave()
+
+    def _replace_ocio_path_with_env_var(self, path):
+        """ Replace OCIO config path with environment variable
+
+        Environment variable is added as TCL expression to path. TCL expression
+        is also replacing backward slashes found in path for windows
+        formatted values.
+
+        Args:
+            path (str): OCIO config path
+
+        Returns:
+            str: OCIO config path with environment variable
+        """
+        # QUESTION: should we also include other names variants
+        included_vars = [
+            "BUILTIN_OCIO_ROOT",
+            "OPENPYPE_PROJECT_ROOT"
+        ]
+        for env_var, env_path in os.environ.items():
+            # first check if variable is whitelisted
+            if all(var_ not in env_var for var_ in included_vars):
```

Comment:
Are we expecting the included vars to be a PART of the environment key names? Could you provide an example of what names of vars we'd be matching?

Reply:
Ah, `OPENPYPE_PROJECT_ROOT_RESOURCES` seems to be one of them.